### PR TITLE
Allow un and re-masking of IRQs.

### PR DESCRIPTION
**Pull request by terminal-cs**
Title explains it.

### CODE REVIEW

**Review by AnErrupTion — CHANGES_REQUESTED** [review]
You'll have to pull from master and change the IOPort read/writes.

File: source/Cosmos.Core/INTs.cs [hunk 1]
```diff
@@ -239,6 +239,40 @@ public struct IRQContext {
...
+
+            if (aIRQLine < 8)
+            {
+                // Read PIC1 Data
+                Port = IOPort.Read8(0x20);
+            }
+            else
+            {
+                // Read PIC2 Data
+                Port = IOPort.Read8(0xA0);
```
Comment (AnErrupTion) [inline source/Cosmos.Core/INTs.cs:260]: According to https://wiki.osdev.org/8259_PIC, if you want to read PIC1 data then Port should be 0x21, and if you want to read PIC2 data it should be 0xA1. See section "ISR and IRR", where PIC1_DATA and PIC2_DATA are defined.

**Review by AnErrupTion — CHANGES_REQUESTED** [review]

File: source/Cosmos.Core/INTs.cs (hunk 1 above)
Comment (terminal-cs) [reply source/Cosmos.Core/INTs.cs:260]: mybad

Comment (AnErrupTion) [reply source/Cosmos.Core/INTs.cs]: @terminal-cs You shouldn't read from those ports. ``Port`` should directly equal to 0x21 for PIC1 and 0xA1 for PIC2.

**Comment by quajak** [comment]
Is there any way we could test if the enabling/disabling works in TestRunner?

**Comment by AsertCreator** [comment]
pit test? we can disable interrupts but unmask pit irq

File: source/Cosmos.Core/INTs.cs (hunk 1 above)
Comment (terminal-cs) [reply source/Cosmos.Core/INTs.cs]: > @terminal-cs You shouldn't read from those ports. `Port` should directly equal to 0x21 for PIC1 and 0xA1 for PIC2.

the port variable is a completely different var, not related to pic1 or 2 addresses

**Comment by terminal-cs** [comment]
> Is there any way we could test if the enabling/disabling works in TestRunner?

i don't know

**Review by quajak — DISMISSED** [review]

**Comment by TheFocusMan** [comment]
> > Is there any way we could test if the enabling/disabling works in TestRunner?
> 
> i don't know

The same problem is in all the PR`s

**Review by quajak — APPROVED** [review]

**Pull request merged**